RocketChat/Rocket.Chat
# [NEW] Incompatible Apps

## Proposed changes (including videos or screenshots)
This PR is introducing incompatible apps to the marketplace. 

Right now, if an app or an app version is incompatible with the current version of Rocket.Chat the user is using, we do not return them to the marketplace. We are changing that to allow users to see the apps and updates they are missing when they do not update their Rocket.Chat version.

The incompatibility is decided by looking at the Apps Engine version of the Rocket.Chat instance is running and matching with the versions of Apps Engine an app supports.

Take a look at this diagram to understand how we match versions.

![versions_matching](https://user-images.githubusercontent.com/4161171/202534132-70c8de63-2f5f-4d2a-b22d-6d87acbfb891.png)




## Issue(s)

[MKP-138](https://rocketchat.atlassian.net/browse/MKP-138)
[MKP-133](https://rocketchat.atlassian.net/browse/MKP-133)
[MKP-133](https://rocketchat.atlassian.net/browse/MKP-135)
[MKP-106](https://rocketchat.atlassian.net/browse/MKP-106)


## Steps to test or reproduce
In order to provide backward compatibility with previous versions of Rocket.Chat, we check which version of apps engine is a Rocket.Chat user using to either return the incompatibles or not. After this Incompatible feature got merged into production, we will update the minimum version on the marketplace API.

That said, to test, you need to override the Apps Engine version you are going to send to the marketplace. Change the value of appsEngineVersionForMarketplace at https://github.com/RocketChat/Rocket.Chat/blob/develop/apps/meteor/app/apps/server/communication/rest.js. For instance, change it to 1.37.0.

Now, if you look at marketplace apps list, probably all apps are incompatible and you can test the whole feature.



## Further comments

Some extra tasks included:
* Migrated AppStatus from JavaScript to TypeScript
* Added TooltipOnHover component to reuse some code
* Fixed marketplace app details page version (it was showing the installed version number)
